micov/tests/test_io.py

```diff
@@ -291,6 +292,76 @@ def test_parse_genome_lengths_bad_sizes(self):
         with self.assertRaisesRegex(ValueError, "Lengths of zero or less"):
             parse_genome_lengths(self.name)
 
+    def test_parse_taxonomy_good(self):
+        data = ("genome_id\ttaxonomy\tbaz\n"
+                "a\tspecies1\txyz\n"
+                "b\tspecies2\txyz\n"
+                "c\tspecies3\txyz\n")
+        
+        with open(self.name, 'w') as fp:
+            fp.write(data)
+
+        exp = pl.DataFrame([['a', "species1"],
+                            ['b', "species2"],
+                            ['c', "species3"]],
+                           schema=[COLUMN_GENOME_ID, COLUMN_TAXONOMY])
+        obs = parse_taxonomy(self.name)
+        plt.assert_frame_equal(obs, exp)
+
+    def test_parse_taxonomy_noheader(self):
+        data = ("a\tspecies1\txyz\n"
+                "b\tspecies2\txyz\n"
+                "c\tspecies3\txyz\n")
+        
+        with open(self.name, 'w') as fp:
+            fp.write(data)
+
+        exp = pl.DataFrame([['a', "species1"],
+                            ['b', "species2"],
+                            ['c', "species3"]],
+                           schema=[COLUMN_GENOME_ID, COLUMN_TAXONOMY])
+        obs = parse_taxonomy(self.name)
+        plt.assert_frame_equal(obs, exp)
+
+    def test_set_taxonomy_as_id(self):
+        cov = pl.DataFrame({
+            "genome_id": ["G000006925", "G000007525", "G000008865"],
+            "covered": [2501356, 4378, 2582128],
+            "length": [4828820, 2260266, 5594477],
+            "percent_covered": [51.800564112971706, 0.19369401654495533, 46.154948889771106]
+            })
+
+        tax = pl.DataFrame({
+            "genome_id": ["G000009925", "G000010425", "G000006925", "G000007525", "G000008865"],
+            "taxonomy": ["species1", "species2", "species3", "species4", "species5"]
+            })
+
+        exp = pl.DataFrame({
```

Comment:
See comment above -- I think we do need to keep the genome ID. We're entering into a position though where we have similar but not identical schemas, so we may need to start rethinking how they're expressed and organized.

Reply:
agreed. Will keep genomeID in the next commit.

Reply:
I think this comment was addressed? If so could "resolve conversation" be used?